pkg/controllers/monitoring/monitoring-stack/components.go

```diff
@@ -215,11 +243,48 @@ func newPrometheus(
 				},
 			},
 		}
+		if ms.Spec.AlertmanagerConfig.WebTLSConfig != nil {
+			caSecret := ms.Spec.AlertmanagerConfig.WebTLSConfig.CA
+
+			prometheus.Spec.Alerting.Alertmanagers[0].Scheme = "https"
+			// NOTE: The following doesn't add the CA cert to the tls-assets secret, so it never gets mounted.
+			// Either I'm missing some configuration or the p-o fork has a bug or it's an upstream bug.
+			// I didn't find a mention of a similar bug anywhere though.
+			/*			
+			prometheus.Spec.Alerting.Alertmanagers[0].TLSConfig = &monv1.TLSConfig{
+				SafeTLSConfig: monv1.SafeTLSConfig{
+					CA: monv1.SecretOrConfigMap{
+						Secret: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: alertmanagerTLSSecretName,
+							},
+							Key: TLSCaKey,
+						},
+					},
+					ServerName: ms.Name + "-alertmanager",
+				},
+			}
```

Comment:
Is this a known bug? I didn't find anything about it. Is this happening only in the forked operator or is it the same upstream?

Reply:
can you try to repro with the upstream operator?

Reply:
I'm adding this to my personal TODO list and if I see the same behavior with the upstream operator, I'll file a bug. In the meantime I'll remove this commented section, I left it there just to show you what isn't working.

Reply:
I replicated the issue with the upstream operator from main branch and created an issue: https://github.com/prometheus-operator/prometheus-operator/issues/6760